core/src/main/scala/filodb.core/memstore/PartKeyLuceneIndex.scala

```diff
@@ -995,7 +995,7 @@ class CardinalityCountBuilder(partSchema: PartitionSchema, cardTracker: Cardinal
         binaryValue.bytes, unsafePkOffset, partSchema.options.shardKeyColumns)
 
       // update the cardinality count by 1, since the shardKey for each document in index is unique
-      cardTracker.modifyCount(shardKey, 1, 1)
```

Comment:
Confirming this was intended for this PR?

Reply:
Can the Builder accept the deltas? That way we can reuse the collector if needed